internal/kconfig/kconfig.go

```
@@ -263,6 +263,21 @@ func PutInteger(data []byte, integer *btf.Int, n uint64) error {
 		return fmt.Errorf("invalid boolean value: %d", n)
 	}
 
+	if len(data) < int(integer.Size) {
+		return fmt.Errorf("byte array size %d is less than the integer size %d", len(data), integer.Size)
```

Nit: technically, data is a byte slice, not an array. Also, it's always better to tell the user what the problem is. Sure the byte slice is smaller, but why is that an issue?
```suggestion
		return fmt.Errorf("can't fit an integer of size %d into a byte slice of length %d", integer.Size, len(data))
```